File: Python/Math-Array-String/82.py
# 3005. Count Elements With Maximum Frequency

from collections import Counter
from typing import List

class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        # Step 1: Count the frequency of each element
        freq = Counter(nums)
        
        # Step 2: Find the maximum frequency
        max_freq = max(freq.values())
        
        # Step 3: Sum the frequencies of elements with the maximum frequency
        total = sum(count for count in freq.values() if count == max_freq)
        
        return total
    
sol = Solution()
output = sol.maxFrequencyElements([1,1,3,2,3])
print(output)  # Output: 4

File: JavaScript/Math-Array-String/82.js
// 3005. Count Elements With Maximum Frequency


/**
 * @param {number[]} nums
 * @return {number}
 */
var maxFrequencyElements = function(nums) {
    // Step 1: Count the frequency of each element
    let freqMap = new Map();
    for (let num of nums) {
        freqMap.set(num, (freqMap.get(num) || 0) + 1);
    }

    // Step 2: Find the maximum frequency
    let maxFreq = Math.max(...freqMap.values());

    // Step 3: Sum the frequencies of elements with the maximum frequency
    let total = 0;
    for (let freq of freqMap.values()) {
        if (freq === maxFreq) {
            total += freq;
        }
    }

    return total;
};

// Test Cases
console.log(maxFrequencyElements([1, 2, 2, 3, 1, 4])); // Output: 4
console.log(maxFrequencyElements([1, 2, 3, 4, 5]));    // Output: 5
